packages/iov-bns/src/decode.ts

```diff
@@ -23,11 +23,22 @@ import {
 import { Encoding } from "@iov/encoding";
 
 import * as codecImpl from "./generated/codecimpl";
-import { asNumber, decodeFullSig, ensure } from "./types";
+import { asNumber, BnsAddressBytes, BnsUsernameNft, decodeFullSig, ensure } from "./types";
 import { encodeBnsAddress, isHashIdentifier } from "./util";
 
 const { fromUtf8 } = Encoding;
 
+export function decodeUsernameNft(nft: codecImpl.username.IUsernameToken): BnsUsernameNft {
+  return {
+    id: fromUtf8(nft.base!.id!),
```

Comment:
What I often do in these cases, where the data comes from outside, is to use `ensure`, which will throw a more useful error than just `.id!`. (or at least should throw a better error)

Reply:
ex: `const sig = ensure(sigs[0], "first signature");`

So error is "first signature is missing", not "cannot access id of undefined"

Reply:
Thanks, right. Let's do what we always do. Will update for both decodeBlockchainNft/decodeUsernameNft in the other PR